UnityProject/Storybook/Assets/Code/Scripts/Game/Player/PlayerMover.cs

```diff
@@ -0,0 +1,338 @@
+            m_worldPlayers[0].TargetNode = m_currentRoom.Player1Node;
+            m_worldPlayers[0].IsAtTarget = false;
+        }
+        if (m_worldPlayers.Count >= 2)
+        {
+            m_worldPlayers[1].TargetNode = m_currentRoom.Player2Node;
+            m_worldPlayers[1].IsAtTarget = false;
+        }
+        if (m_worldPlayers.Count >= 3)
+        {
+            m_worldPlayers[2].TargetNode = m_currentRoom.Player3Node;
+            m_worldPlayers[2].IsAtTarget = false;
+        }
+        if (m_worldPlayers.Count >= 4)
+        {
+            m_worldPlayers[3].TargetNode = m_currentRoom.Player4Node;
+            m_worldPlayers[3].IsAtTarget = false;
+        }
+    }
+
+    /// <summary>
+    /// Moves the camera to the new position and rotation
+    /// Called when the players switch rooms
+    /// </summary>
+    /// <param name="newCamPos">The new position for the camera</param>
+    /// <param name="newCamRot">The new rotation for the camera</param>
+    [PunRPC]
+    public void SwitchCameraRoom(Vector3 newCamPos, Quaternion newCamRot)
+    {
+        Camera.main.transform.position = newCamPos;
+        Camera.main.transform.rotation = newCamRot;
+    }
+
+    /// <summary>
+    /// Sends the current room data to the clients
+    /// </summary>
+    /// <param name="isNorthDoorActive">True if the north door is active for the current room, false otherwise</param>
+    /// <param name="isEastDoorActive">True if the east door is active for the current room, false otherwise</param>
+    /// <param name="isSouthDoorActive">True if the south door is active for the current room, false otherwise</param>
+    /// <param name="isWestDoorActive">True if the west door is active for the current room, false otherwise</param>
+    /// <param name="roomType">The type of the room (MapManager.RoomType)</param>
+    [PunRPC]
+    public void SendRoomData(bool isNorthDoorActive, bool isEastDoorActive, bool isSouthDoorActive, bool isWestDoorActive, int roomType)
+    {
+        m_currentRoomData = new RoomData(isNorthDoorActive, isEastDoorActive, isSouthDoorActive, isWestDoorActive, (MapManager.RoomType) roomType);
+    }
+
+    /// <summary>
+    /// Sends the location of the current room to the clients
+    /// </summary>
+    /// <param name="roomLocX">The x-coordinate of the current room</param>
+    /// <param name="roomLocY">The y-coordinate of the current room</param>
+    [PunRPC]
+    public void SendRoomLoc(int roomLocX, int roomLocY)
+    {
+        m_currentRoomLoc = new Location(roomLocX, roomLocY);
+    }
+
+    /// <summary>
+    /// Called on clients to create a room on the server
+    /// </summary>
+    /// <param name="roomLocX">The x-coordinate of the room to create</param>
+    /// <param name="roomLocY">The y-coordinate of the room to create</param>
+    /// <param name="doorIndex">The door that was selected by the client</param>
+    [PunRPC]
+    public void CreateRoomOnMaster(int roomLocX, int roomLocY, int doorIndex)
```

Comment:
Don't use "Master" use owner, owner is more general and flexible than master. The master client may not be the owner.